client_plugin/drivers/shared/etcdops.go

```diff
@@ -72,10 +77,32 @@ type kvPair struct {
 	value string
 }
 
+type kvStore interface {
+	// Init - initialize the KV store cluster
+	Init() error
+	// WriteVolMetadata - Update or Create volume metadata in KV store
+	WriteVolMetadata(entries []kvPair) error
+	// ReadVolMetadata - Read volume metadata in KV store
+	ReadVolMetadata(keys []string) ([]kvPair, error)
+	// DeleteVolMetadata - Delete volume metadata in KV store
+	DeleteVolMetadata(name string) error
+	// CompareAndPut - Compare the value of key with oldVal, if equal, replace with newVal
+	CompareAndPut(key string, oldVal string, newVal string) bool
+	// ListVolumeName - List all the volumes in the KV store
+	ListVolumeName() ([]string, error)
+}
+
+type etcdInfo struct {
```

Comment:
Can we have a better name for this interface? The name "etcdInfo" sounds like a data object.